File: fix_encoding.js
const { execSync } = require('child_process');
const fs = require('fs');

// Get clean App.jsx from git as a Buffer (raw bytes)
const appJsx = execSync('git show 958b87c:src/App.jsx');
fs.writeFileSync('src/App.jsx', appJsx);
console.log('OK -', appJsx.length, 'bytes written to src/App.jsx');

// Get clean App.css from git
const appCss = execSync('git show 958b87c:src/App.css');
fs.writeFileSync('src/App.css', appCss);
console.log('OK -', appCss.length, 'bytes written to src/App.css');

File: fix_encoding.py
import subprocess
import os

# Get the clean file from git as raw bytes (UTF-8 without BOM)
content = subprocess.check_output(['git', 'show', '958b87c:src/App.jsx'])

# Write raw bytes directly - preserves exact UTF-8 encoding from git
with open('src/App.jsx', 'wb') as f:
    f.write(content)

print('OK -', len(content), 'bytes written to src/App.jsx')

# Also restore App.css
content_css = subprocess.check_output(['git', 'show', '958b87c:src/App.css'])
with open('src/App.css', 'wb') as f:
    f.write(content_css)

print('OK -', len(content_css), 'bytes written to src/App.css')
